src/ops/info.rs

```diff
@@ -47,6 +48,13 @@ pub fn info(
         package_id = None;
     }
 
+    let current_rustc_version = config.load_global_rustc(ws.as_ref())?.version;
+    let current_rustc_version = &semver::Version::new(
+        current_rustc_version.major,
+        current_rustc_version.minor,
+        current_rustc_version.patch,
+    );
```

Comment:
why is this being done?

Reply:
Oh. it was a mistake. We don't need to clean up pre-release identifiers here.

Reply:
I tested it again. I found if we don't ignore the build and pre-release information. Then the version `rustc 1.76.0-nightly (21cce21d8 2023-12-11)` can not matched by `1.1.0`.

Reply:
The beta version also can not matched by `1.0.0`.

Reply:
... I hope I didn'r break anything within cargo related to this.

With regards to this code, I can think of two directions to go
- Just don't do the check
- Check for the prior stable release

Reply:
> * Just don't do the check

Do you mean we ignore pre-release identifiers here or do not check the current local rustc version at all?

Reply:
One option is we just ignore `rustc --version` if its prerelease.

Reply:
How about we ignore the pre-release and build information? Because we don't allow `1.73.0-nightly` in the `Cargo.toml`. So maybe we should simply ignore the pre-release and build information. I think `cargo install` command also follows the same rule.

Reply:
I agree about matching existing behavior. I do need to look over it in general to see if it makes sense.